keyboards/ergodox_ez/keymaps/colemak_es_osx/keymap.c

```diff
@@ -0,0 +1,405 @@
+ *                                 |      |      |      |       |      |      |      |
+ *                                 |      |      |------        |------|      |      |
+ *                                 |      |      |      |       |      |      |      |
+ *                                 `--------------------'       `--------------------'
+ */
+[IDEA] = LAYOUT_ergodox(
+
+//Left hand
+        _______,        _______,    _______,    _______,    _______,    _______,    I_RENAME,
+        _______,        _______,    _______,    I_SEARCH,   _______,    _______,    _______,
+        _______,        _______,    _______,    _______,    _______,    _______,
+        _______,        _______,    P_CODE,     _______,    _______,    _______,    _______,
+        _______,        _______,    _______,    _______,    _______,
+
+                                                                        _______,    DF(BASE),
+                                                                                    _______,
+                                                        _______,        _______,    _______,
+
+// right hand
+        I_FUSAGE,       _______,    I_LN_COM,   I_BK_COM,   _______,    I_TERM,    _______,
+        I_COLUMN,       _______,    I_REFORM,   _______,    _______,    _______,    _______,
+                        _______,    _______,    _______,    _______,    _______,    _______,
+        _______,        _______,    _______,    _______,    _______,    _______,    _______,
+                                _______,        _______,    _______,    _______,    _______,
+
+        _______,        _______,
+        _______,
+        _______,        _______,    _______
+)
+};
+
+
+void led_1_off(void) {
+    ergodox_right_led_1_off();
+}
+
+void led_2_off(void) {
+    ergodox_right_led_2_off();
+}
+
+void led_3_off(void) {
+    ergodox_right_led_3_off();
+}
+
+//Runs just one time when the keyboard initializes
+void matrix_init_use(void) {
+    led_1_off();
+    led_2_off();
+    led_3_off();
+}
+
+// Value to use to switch LEDs on. The default value of 255 is far too bright.
+static const uint8_t max_led_value = 20;
+
+
+void led_1_on(void) {
+    ergodox_right_led_1_on();
+    ergodox_right_led_1_set(max_led_value);
+}
+
+void led_2_on(void) {
+    ergodox_right_led_2_on();
+    ergodox_right_led_2_set(max_led_value);
+}
+
+void led_3_on(void) {
+    ergodox_right_led_3_on();
+    ergodox_right_led_3_set(max_led_value);
+}
+
+static uint32_t current_layer_state = 0;
+// Is the layer momentarily active?
+#define M_LAYER_ON(layer) (current_layer_state & (1<<layer))
+
+//
+uint32_t layer_state_set_user(uint32_t state) {
+    current_layer_state = state;
+
+    if(M_LAYER_ON(IDEA)) {
+        led_1_on();
+        led_3_on();
+    } else {
+        led_1_off();
+        led_3_off();
+    }
+
+  return state;
+};
+
+#define LAYER_ON(layer) (default_layer_state & (1<<layer))
```

Comment:
This isn't actually needed. 
```suggestion
```

Specifically, you may want to use `layer_state_cmp`, as that handles this. 

You'd want to call `layer_state_cmp(default_layer_state, LAYER)`. 

https://github.com/qmk/qmk_firmware/blob/ba26736d7e0679e87e34add2445cdae541e2a0b8/tmk_core/common/action_layer.c#L154-L157

Reply:
Fixed.
Thanks for the tip